tests/binning_test.py

```diff
@@ -55,7 +55,27 @@ def test_hist_table_define_edges_from_bin_count():
     assert histogrammed.sizes['y'] == 4
     assert edges.min().value == da.coords['y'].min().value
     assert edges.max().value > da.coords['y'].max().value
-    assert edges.max().value == np.nextafter(da.coords['y'].max().value, np.inf)
+    ymax = da.coords['y'].max()
+    assert edges.max().value == np.nextafter(
+        ymax.value, (ymax + sc.scalar(1.0, unit=ymax.unit, dtype=ymax.dtype)).value)
+
+
+def test_hist_table_define_edges_from_bin_count_float32():
+    da = sc.data.table_xyz(100)
+    for c in list(da.coords.keys()):
+        da.coords[c] = da.coords[c].to(dtype='float32')
+    # Make the last y value larger, so we end up with only one element in the last bin
+    da.coords['y'].values[-1] = 5.0
```

Comment:
Why is this needed? I am not sure why you did not simply parametrize the existing test above for float32 and float64.